test/static/reject/ufuncs.pyi

```diff
@@ -46,3 +47,8 @@ np.isnan(i8, dtype=int)  # type: ignore[arg-type]  # pyright: ignore[reportArgum
 np.isnat(i8, dtype=np.int64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
 np.isinf(i8, dtype=np.int64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
 np.isfinite(i8, dtype=np.int64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
+
+np.logical_not(dt64, dtype=np.datetime64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
+np.logical_and(dt64, dt64, dtype=np.datetime64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
+np.logical_or(dt64, dt64, dtype=np.datetime64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
+np.logical_xor(dt64, dt64, dtype=np.datetime64)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType, reportCallIssue]
```

Comment:
quick nitpick: you're testing for two illegal things at the same time here; passing `datetime64` scalars, and passing `type[datetime64]` as `dtype`. It's usually recommended to test 1 thing at the same time. That makes it easier to debug when things go wrong.